src/operator/tensor/dot-inl.h

```diff
@@ -613,15 +613,15 @@ inline void DotCsrDnsRspImpl(const OpContext& ctx,
       MSHADOW_IDX_TYPE_SWITCH(col_idx_l.type_flag_, CType, {  // col idx type
         MSHADOW_IDX_TYPE_SWITCH(ret->aux_type(rowsparse::kIdx), RType, {  // row idx type
           const dim_t num_rows = lhs.shape()[1];
-          size_t workspace_size = 2 * (num_rows * sizeof(dim_t));
+          size_t workspace_size = num_rows * sizeof(dim_t);
           mshadow::Tensor<cpu, 1, char> workspace =
             ctx.requested[0].get_space_typed<cpu, 1, char>(
             mshadow::Shape1(workspace_size), s);
           dim_t* row_flg = reinterpret_cast<dim_t*>(workspace.dptr_);
-          dim_t* prefix_sum = row_flg + num_rows;
-
-          Fill<false>(s, TBlob(row_flg, mshadow::Shape1(num_rows), cpu::kDevMask), kWriteTo, 0);
-          mxnet_op::Kernel<MarkRowFlgKernel, cpu>::Launch(s, lhs.aux_shape(csr::kIdx)[0], row_flg,
+          // prefix sum array re-uses the row_flg array temp space
+          dim_t* prefix_sum = row_flg;
+          Kernel<op_with_req<set_to_int<0>, kWriteTo>, cpu>::Launch(s, num_rows, row_flg);
```

Comment:
Why not use `set_zero`? Or should `set_zero` in line 643 also be replaced by `op_with_req<set_to_int<0>, kWriteTo>`?

Reply:
set_zero is just a 'using' of set_to_int<0>, so it's the same thing.

Reply:
I am curious, why was Fill() removed?  Fill() will be faster on CPU than set_to_int<0> or set_zero in many cases (in almost all but the most extremely large tensor cases, actually).

Reply:
The number of elements in this array is in the order of millions. I tried 2M & 20M, `set_to_int` is several times faster than Fill. What cases did you test? Maybe we should add a switch to `Fill` depending on the size of the tensor